CONTRIBUTING.md

```diff
@@ -2,32 +2,97 @@
+
+Now you have a local directory of the git repository. Any changes to the contents of the directory can now be tracked using git. But before you make your desired change, we recommend creating a branch. When you clone a repository, you will, by default, be on the `master` branch. However, you don't want to make changes on your `master` branch, since it's best to keep it synced with the `upstream` repository. Therefore, you will create a new branch for making your change. So say I wanted to add my name to `cognoma/sandox`, I would create a new branch named `add-username` with the following command:
+
+```sh
+# Create a new branch named add-username and switch to it
+git checkout --branch add-username
+```
+
+Now you will be on your `add-username` branch. You should now make your change, which in this example will be adding a new file but can also be modifying an existing file. Let's say you create the file `people/username.md` in your text editor of choice. Now you will have to stage your change by running:
+
+```sh
+# Show what changes git detects.
+# Not essential but good practice as it may alert you to an issue.
+git status
+
+# Add the file you've created
+git add people/username.md
+```
 
-***
+Next you will have to make a commit, which takes a snapshot of everything that's been staged (using `git add` above). To commit, run `git commit`. A text editor should appear that allows you to write a [commit message](https://chris.beams.io/posts/git-commit/) where you should describe the changes in the commit. If all succeeds, you will have added to the git repository and your commit will be visible when you run `git log`.
 
-_Note: You do not have to read the following information to contribute._
+Next you will push your commit to GitHub, which means add it to `username/sandbox` on GitHub. To do this, run:
 
-## Commit Access
+```sh
+# This will only have to be run once per computer
+# It adopts the Git 2.0, removing a confusing message
+git config --global push.default simple
+
+# Push your current branch to GitHub
+# If this is the first time you're pushing this branch,
+# you will receive an error suggesting you specify --set-upstream.
+# If so, use the suggested command. Then future git push attempts will work.
+git push
+```
+
+Now you are ready to create your [pull request](https://help.github.com/articles/about-pull-requests/). The GitHub webpage for either `cognoma/sandbox` or `username/sandbox` should have a new button to "Compare & pull request." Click it and write a title and description for your requested changes. Then submit. Soon (hopefully), another Cognoma contributor will review your pull request. Usually, pull requests will require some changes before they are ready to merge (incorporate into Cognoma repository). You can add additional changes to a pull request by adding commits to the branch its based on, which is this example is `add-username`.
```

Comment:
any thoughts on adding PR and [issue templates](https://github.com/blog/2111-issue-and-pull-request-templates)? I've added them to some of my projects and have been happy with them so far